### ISSUE TITLE
Unnecessarily large padding

### ISSUE THREAD

**Issue by thomasaarholt** [matplotlib/jupyter-matplotlib#72]
`%matplotlib widget` introduces a lot of padding around the plotted image, emphasised when compared to `%matplotlib inline`
I have tried various matplotlib calls, but can't figure out a way to reduce it. In any case, I think it should be default be smaller than it is? 
I'm sure I'm not the only one who's had to scroll a bit too much to get past very large images.

Two additional questions:

- what controls the increased size of the plot with `widget`?
- Can we reduce or change the size/shape of the "Figure 1" toolbar? I understand the need to close the interactive plot, but do we need the full toolbar?

![image](https://user-images.githubusercontent.com/2721423/48932234-0f8e8300-eefb-11e8-89f3-c0b6f10f3235.png)
![image](https://user-images.githubusercontent.com/2721423/48932255-2d5be800-eefb-11e8-94dd-0ba1c6dcaafd.png)

Test code:
```
%matplotlib widget
import numpy as np
import matplotlib.pyplot as plt
plt.figure()
plt.imshow(np.random.random((32,32)));
```

**Comment by thomasaarholt** [comment]
The padding seems particularly unnecessary with `plt.subplots()`:
![image](https://user-images.githubusercontent.com/2721423/48932424-22558780-eefc-11e8-8aba-84d1d9a0253d.png)

**Comment by hadim** [comment]
I agree with this!

**Comment by coolsurf** [comment]
Any suggestions or workaround?  Thanks.

**Comment by GarrisonD** [comment]
I agree with this too! Any updates? @martinRenou

**Comment by GarrisonD** [comment]
@martinRenou As far as I understand after reading the source code, `ipympl` has nothing to do with this issue since it's just fetching the image from the backend and rendering on the canvas without any space being added. So the problem is on the backend side, right?

**Comment by martinRenou** [comment]
> As far as I understand after reading the source code, ipympl has nothing to do with this issue since it's just fetching the image from the backend and rendering on the canvas without any space being added. So the problem is on the backend side, right?

Yeah you're right. Although the header/footer and the toolbox are taking some space. But they can still be hidden so there should not be any problem.

Actually this issue starts to be a bit old, and lots of changes have happened since those screenshots were taken. Shouldn't we close this issue? Or did you find that ipympl was still taking too much space with recent releases @GarrisonD ?

**Comment by GarrisonD** [comment]
I am using the latest version (0.5.6) and can confirm that there is still too much space around. For some cases `plt.tight_layout()` helps but specified `bbox_inches='tight'` is more robust but I can't figure out how to use that with `ipympl` 😢 

With `inline`:

<img width="448" alt="Screenshot 2020-05-13 at 16 30 52" src="https://user-images.githubusercontent.com/13675895/81818869-40bba280-9537-11ea-8d87-58a1ff9dc85a.png">

With `widget`:

<img width="753" alt="Screenshot 2020-05-13 at 16 31 05" src="https://user-images.githubusercontent.com/13675895/81818877-43b69300-9537-11ea-98fd-30a6e5873724.png">

With `widget` and `tight layout`:

<img width="748" alt="Screenshot 2020-05-13 at 16 32 56" src="https://user-images.githubusercontent.com/13675895/81819034-782a4f00-9537-11ea-944a-c813dc8f301e.png">

**Comment by GarrisonD** [comment]
@martinRenou Any ideas on how to pass `bbox_inches` to render process?

**Comment by GarrisonD** [comment]
@martinRenou `matplotlib.backend_bases.FigureCanvasBase.print_figure` accepts `bbox_inches` option and is used in `InlineBackend` (see [source](https://github.com/ipython/ipython/blob/master/IPython/core/pylabtools.py#L132)).
Is it possible for `nbAgg` backend to work this way?

**Comment by martinRenou** [comment]
Maybe @tacaswell you would have some comments? I am not familiar with the Matplotlib API.

**Comment by ianhi** [comment]
On ipympl `0.6.2` and matplotlib `'3.3.2.post2048+g448100ae0'` I believe that ipympl is doing the correct thing - giving the same result as other interactive backends. Here is a comparison showing that ipympl produces the same padding as the `TkAgg` backend.
![image](https://user-images.githubusercontent.com/10111092/103727094-6571f800-4fa8-11eb-8f4b-5f529dd70b51.png)

The `inline` backend is essentially calling `tight_layout` because the default argument to `print_figure` is `bbox_inches='tight'`
https://github.com/ipython/ipython/blob/3587f5bb6c8570e7bbb06cf5f7e3bc9b9467355a/IPython/core/pylabtools.py#L97


To get permanent results consider this quote from the [tight_layout tutorial](https://matplotlib.org/3.3.3/tutorials/intermediate/tight_layout_guide.html)

> Note that matplotlib.pyplot.tight_layout() will only adjust the subplot params when it is called. In order to perform this adjustment each time the figure is redrawn, you can call fig.set_tight_layout(True), or, equivalently, set rcParams["figure.autolayout"] (default: False) to True.

**Comment by ianhi** [comment]
> that there is still too much space around. 

I think the place to take this up would be on the matplotlib repo as this isn't something ipympl controls.

**Comment by marscher** [comment]
Could it be, that we have some kind of regression here?
![grafik](https://github.com/matplotlib/ipympl/assets/170287/8f367e1a-9e4d-4e90-a142-9947dc35b813)
The top and bottom padding are very large. The screenshot only shows the top padding, but the lower one is of the same size as the top one.

The code to generate the plot is this snippet:
``` python
plt.figure(figsize=(10, 20))
plt.title("Epipolar lines in rectified images.")
img_epilines2 = np.random.random((1280, 2048, 3))
%matplotlib widget
plt.imshow(img_epilines2)
plt.tight_layout()
plt.show()
```
matplotlib 3.8.4
ipympl 0.9.4

Or dou you think it'd be better to raise a new issue somewhere @martinRenou?

**Comment by martinRenou** [comment]
Yes, probably a new issue would be better 👍🏽

**Comment by marscher** [comment]
xref https://github.com/matplotlib/ipympl/issues/553

**Comment by barsv** [comment]
I came here because I was looking for 

```
fig.canvas.header_visible = False
fig.canvas.footer_visible = False
```

found it in #396
